helm/coredns-app/templates/deployment-masters.yaml

```diff
@@ -55,8 +55,13 @@ spec:
               topologyKey: kubernetes.io/hostname
       nodeSelector:
         {{-  range $key, $value := $.Values.mastersInstance.nodeSelector }}
-        {{ $key }}: {{ $value }}
-        {{- end}}
+          {{- if and (eq $key "node-role.kubernetes.io/control-plane") (lt (int $.Capabilities.KubeVersion.Minor) 25) }}
```

Comment:
Maybe we can use `(semver "1.2.3").Compare` here? https://docs.helm.sh/docs/chart_template_guide/function_list/#semver